src/screens/Profile/TravelToken/index.tsx

```diff
@@ -1,22 +1,46 @@
-import {Edit} from '@atb/assets/svg/mono-icons/actions';
 import FullScreenHeader from '@atb/components/screen-header/full-header';
-import * as Sections from '@atb/components/sections';
-import ThemeText from '@atb/components/text';
-import ThemeIcon from '@atb/components/theme-icon/theme-icon';
-import {useMobileTokenContextState} from '@atb/mobile-token/MobileTokenContext';
 import {StyleSheet, Theme} from '@atb/theme';
 import {TravelTokenTexts, useTranslation} from '@atb/translations';
 import TravelTokenBox from '@atb/travel-token-box';
-import React from 'react';
+import React, {useEffect, useState} from 'react';
 import {View} from 'react-native';
 import {ScrollView} from 'react-native-gesture-handler';
 import {ProfileScreenProps} from '../types';
+import {FaqSection} from '@atb/screens/Profile/TravelToken/FaqSection';
+import {ChangeTokenAction} from '@atb/screens/Profile/TravelToken/ChangeTokenAction';
+import {useMobileTokenContextState} from '@atb/mobile-token/MobileTokenContext';
+import {useIsFocused} from '@react-navigation/native';
 
 type TravelCardScreenProps = ProfileScreenProps<'TravelToken'>;
 
 export default function TravelCard({navigation}: TravelCardScreenProps) {
   const styles = useStyles();
   const {t} = useTranslation();
+  const [showLoader, setShowLoader] = useState<boolean>(false);
+  const [toggleLimit, setToggleLimit] = useState<number | undefined>();
+  const [maxToggleLimit, setMaxToggleLimit] = useState<number | undefined>();
+
+  const {isError, isLoading, getTokenToggleDetails} =
+    useMobileTokenContextState();
+  const screenHasFocus = useIsFocused();
+
+  useEffect(() => {
+    const fetchToggleLimit = async () => {
+      setShowLoader(true);
+      const toggleToggleDetails = await getTokenToggleDetails();
+      if (toggleToggleDetails) {
+        const {toggleMaxLimit, toggledCount} = toggleToggleDetails;
+        if (toggleMaxLimit && toggleMaxLimit >= toggledCount) {
+          setToggleLimit(toggleMaxLimit - toggledCount);
+        }
+        setMaxToggleLimit(toggleMaxLimit);
+      }
+      setShowLoader(false);
+    };
+    if (!isError && !isLoading) {
+      fetchToggleLimit();
+    }
+  }, [getTokenToggleDetails, screenHasFocus]);
```

Comment:
Maybe extract this is a custom hook, making the `TravelTokenScreen` a little less complex, and the developer can themselves figure out if they need/want to deep dive into the custom hook to see the implementation for fetching the toggle details.